src/include/mempool.h

```diff
@@ -376,7 +376,7 @@ class pool_allocator {
                                                                         \
     template<typename k,typename v, typename cmp = std::less<k> >	\
     using map = std::map<k, v, cmp,					\
-			 pool_allocator<std::pair<k,v>>>;		\
+			 pool_allocator<std::pair<const k,v>>>;		\
```

Comment:
@wjwithagen could you add `const` to `multimap` and `unordered_map`'s allocator's key type also? like:

```diff
diff --git a/src/include/mempool.h b/src/include/mempool.h
index 7b4a59618e..ad06fcd769 100644
--- a/src/include/mempool.h
+++ b/src/include/mempool.h
@@ -376,11 +376,11 @@ public:
                                                                         \
     template<typename k,typename v, typename cmp = std::less<k> >      \
     using map = std::map<k, v, cmp,                                    \
-                        pool_allocator<std::pair<k,v>>>;               \
+                        pool_allocator<std::pair<const k,v>>>;         \
                                                                         \
     template<typename k,typename v, typename cmp = std::less<k> >      \
     using multimap = std::multimap<k,v,cmp,                            \
-                                  pool_allocator<std::pair<k,v>>>;     \
+                                  pool_allocator<std::pair<const k,v>>>;       \
                                                                         \
     template<typename k, typename cmp = std::less<k> >                 \
     using set = std::set<k,cmp,pool_allocator<k>>;                     \
@@ -395,7 +395,7 @@ public:
             typename h=std::hash<k>,                                   \
             typename eq = std::equal_to<k>>                            \
     using unordered_map =                                              \
-      std::unordered_map<k,v,h,eq,pool_allocator<std::pair<k,v>>>;     \
+      std::unordered_map<k,v,h,eq,pool_allocator<std::pair<const k,v>>>;       \
                                                                         \
     inline size_t allocated_bytes() {                                  \
       return mempool::get_pool(id).allocated_bytes();                  \
```

and we should use `pair<const Key, T>` as the value type of allocator as per § 23.4.4.1, n3337.

```c++
namespace std {
  template <class Key, class T, class Compare = less<Key>,
            class Allocator = allocator<pair<const Key, T> > >
  class map {
  public:
    // types:
    typedef Key                  key_type;
    typedef T                    mapped_type;
    typedef pair<const Key, T>   value_type;
    // ...
  };
```

the same applies to other associative containers containing key-value pairs. both GCC and Clang comply to this standard. but libstdc++ does not do the `static_assert()`, so we are not aware of this problem until now.